src/main/java/htsjdk/variant/variantcontext/JEXLMap.java

```diff
@@ -11,152 +12,188 @@
 import java.util.Set;
 
 /**
- * this is an implementation of a Map of JexlVCMatchExp to true or false values.  It lazy initializes each value
- * as requested to save as much processing time as possible.
+ * This is an implementation of a Map of {@link JexlVCMatchExp} to true or false values.
+ * It lazily initializes each value as requested to save as much processing time as possible.
  *
+ * TODO: must resolve the following outdated comment in PR review rounds
  * Compatible with JEXL 1.1 (this code will be easier if we move to 2.0, all of the functionality can go into the
  * JexlContext's get()
  *
+ * TODO: there is some troubling design choices in this class: the laziness and exception behavior:
+ * TODO:   because the class is designed to be lazy, when an instance is created, some invalid expressions may be provided
+ * TODO:   later when user requests some/all values, those invalid expressions will result in IllegalArgumentException,
+ * TODO:   especially when calling the function values(), which is the last thing users expect when calling such accessors.
+ * TODO:   but there may be no good choice here. could decide to "unsupport" the values() function.
  */
 
 class JEXLMap implements Map<JexlVCMatchExp, Boolean> {
-    // our variant context and/or Genotype
-    private final VariantContext vc;
-    private final Genotype g;
 
     // our context
     private JexlContext jContext = null;
+    // our variant context and/or Genotype
+    private final VariantContext vc;
+    private final Genotype g;
 
-    // our mapping from JEXLVCMatchExp to Booleans, which will be set to NULL for previously uncached JexlVCMatchExp
+    /**
+     * our mapping from {@link JexlVCMatchExp} to {@link Boolean}s, which will be set to {@code NULL}
+     * for previously un-cached {@link JexlVCMatchExp}.
+     */
     private Map<JexlVCMatchExp,Boolean> jexl;
 
-
-    public JEXLMap(Collection<JexlVCMatchExp> jexlCollection, VariantContext vc, Genotype g) {
+    // -----------------------------------------------------------------------------------------------
```

Comment:
We don't typically have these sorts of big signpost comments.  It seems useful for dividing the unsupported operations from the rest, but I'm not sure you need tthem for the rest of the class.